src/main/java/org/opentripplanner/middleware/controllers/api/ApiController.java

```diff
@@ -188,6 +209,22 @@ private T getObjectForId(Request req, String id) {
         return object;
     }
 
+    /**
+     * Convenience method for extracting the attribute/field param from the HTTP request.
+     */
+    private T getObjectWithField(Request req, String field, String value) {
```

Comment:
Would like this renamed to `getFirstObjectByFieldValue` to indicate that it is getting the first record that matches a field value.